src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -22,428 +22,698 @@ namespace NuGet.Commands
                 {
-                    AddPropsAndTargets(repositories, libraryIdentity, buildItems, result);
+                    doc.Save(output);
                 }
+            },
+            path);
+        }
+
+        public static string GetImportPath(string importPath, string repositoryRoot)
+        {
+            var path = importPath;
+
+            if (importPath.StartsWith(repositoryRoot, StringComparison.Ordinal))
+            {
+                path = $"$(NuGetPackageRoot){importPath.Substring(repositoryRoot.Length)}";
+            }
+            else
+            {
+                path = ReplacePathsWithMacros(importPath);
             }
 
-            return result;
+            return path;
         }
 
-        private static TargetsAndProps GetTargetsAndPropsForCrossTargeting(
-            IEnumerable<RestoreTargetGraph> targetGraphs,
-            IReadOnlyList<NuGetv3LocalRepository> repositories,
-            RemoteWalkContext context,
-            RestoreRequest request,
-            Dictionary<RestoreTargetGraph,
-            Dictionary<string, LibraryIncludeFlags>> includeFlagGraphs)
+        /// <summary>
+        /// Check if the file has changes compared to the original on disk.
+        /// </summary>
+        public static bool HasChanges(XDocument newFile, string path, ILogger log)
         {
-            var result = new TargetsAndProps();
-
-            // Skip runtime graphs, msbuild targets may not come from RID specific packages
-            // Order the graphs by framework to make this deterministic for scenarios where
-            // TFMs disagree on the dependency order, there is little that can be done for
-            // conflicts where A->B for TFM1 and B->A for TFM2.
-            var ridlessGraphs = targetGraphs
-                .Where(g => string.IsNullOrEmpty(g.RuntimeIdentifier))
-                .OrderBy(g => g.Framework, new NuGetFrameworkSorter());
-
-            // Gather props and targets to write out
-            foreach (var graph in ridlessGraphs)
-            {
-                var globalGroupSets = GetMSBuildAssets(
-                    context,
-                    graph,
-                    request.Project,
-                    includeFlagGraphs,
-                    graph.Conventions.Patterns.MSBuildCrossTargetingFiles);
-
-                // Check if compatible build assets exist
-                foreach (var globalGroupEntry in globalGroupSets)
+            if (newFile == null)
+            {
+                // The file should be deleted if it is null.
+                return File.Exists(path);
+            }
+            else
+            {
+                var existing = ReadExisting(path, log);
+
+                if (existing != null)
                 {
-                    var libraryIdentity = globalGroupEntry.Key;
-                    var buildGroupSet = globalGroupEntry.Value;
+                    // Use a simple string compare to check if the files match
```

Comment:
Good call. This didn't change in this PR, it just moved. But now that contentFiles could potentially make these files larger proper hashing would be nice. https://github.com/NuGet/Home/issues/4120